bson/src/test/unit/org/bson/types/ObjectIdTest.java

```diff
@@ -42,6 +44,28 @@ public void testToBytes() {
 
     @Test
     public void testFromBytes() {
+
+        try {
+            new ObjectId((byte[]) null);
+            fail("Expected IllegalArgumentException");
+        } catch (IllegalArgumentException e) {
+            assertEquals("bytes can not be null", e.getMessage());
+        }
+
+        try {
+            new ObjectId(new byte[11]);
+            fail("Expected IllegalArgumentException");
+        } catch (IllegalArgumentException e) {
+            assertEquals("state should be: bytes has length of 12", e.getMessage());
```

Comment:
I still think we should look to improve the error messages.  "state should be: ... " isn't the friendliest error message.